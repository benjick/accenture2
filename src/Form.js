import React, {Component} from 'react';

class Form extends Component {
  submit = event => {
    event.preventDefault();
    this.props.sendMessage();
  };
  render() {
    return (
      <form onSubmit={this.submit}>
        <div style={{flex: 0, height: 30, display: 'flex'}}>
          <input
            style={{height: '100%'}}
            onChange={e => this.props.setName(e.target.value)}
            value={this.props.name}
          />
          <input
            style={{flex: 1, height: '100%'}}
            onChange={e => this.props.setMessage(e.target.value)}
            value={this.props.message}
          />
          <button>Send message</button>
        </div>
      </form>
    );
  }
}

export default Form;
