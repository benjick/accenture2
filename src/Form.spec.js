import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Form from './Form';

describe('Form Component', () => {
  it('should do something', () => {
    const cb = jest.fn();
    const component = mount(<Form sendMessage={cb} />);
    component.find('.qa-name').simulate('change', {
      target: {
        value: 'Irene',
      },
    });
    component.find('.qa-message').simulate('change', {
      target: {
        value: 'Hello2',
      },
    });
    component.find('form').simulate('submit');
    expect(cb.mock.calls.length).toBe(1);
    expect(cb.mock.calls[0][0]).toBe('Irene');
    expect(cb.mock.calls[0][1]).toBe('Hello2');
  });

  it('should render the same snapshot every time', () => {
    const cb = jest.fn();
    const tree = renderer.create(<Form sendMessage={cb} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
