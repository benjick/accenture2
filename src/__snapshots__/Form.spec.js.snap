// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render the same snapshot every time 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 0,
        "height": 30,
      }
    }
  >
    <input
      className="qa-name"
      onChange={[Function]}
      style={
        Object {
          "height": "100%",
        }
      }
      value="Max"
    />
    <input
      className="qa-message"
      onChange={[Function]}
      style={
        Object {
          "flex": 1,
          "height": "100%",
        }
      }
      value="Hello"
    />
    <button>
      Send message
    </button>
  </div>
</form>
`;
